lib/poller.js

```diff
@@ -209,7 +209,12 @@ class Poller {
     // Testing data property
     if (namingConvention && externalData) {
       externalData.forEach((secretProperty, index) => {
-        const reNaming = new RegExp(namingConvention)
+        let reNaming = new RegExp()
+        if (Array.isArray(namingConvention)) {
+          reNaming = new RegExp(namingConvention.join('|'))
+        } else {
+          reNaming = new RegExp(namingConvention)
+        }
```

Comment:
Maybe we can move this out of the loop so its only done once eg moving it after `const namingConvention = namespace.metadata.annotations[this._namingPermittedAnnotation]`, should be fine even if we there's no `namingConvention` set. Keeping the `if` checks the same 

Then we also shouldn't need to repeat the logic for `dataFrom` below

Reply:
Yes, totally. I was mislead by the `const reNaming = new RegExp(namingConvention)` in the `forEach` loop.